docs/customization/customizing-emails.mdx

```diff
@@ -0,0 +1,174 @@
+
+# frozen_string_literal: true
+
+module AmazingStore
+  class CartonMailer < Spree::BaseMailer
+    # Send an email to customers to notify that an individual carton has been
+    # shipped. If a carton contains items from multiple orders then this will be
+    # called with that carton one time for each order.
+    #
+    # This custom version also retrieve the estimated time of arrival.
+    #
+    # @option options carton [Spree::Carton] the shipped carton
+    # @option options order [Spree::Order] one of the orders with items in the carton
+    # @option options resend [Boolean] indicates whether the email is a 'resend' (e.g.
+    #   triggered by the admin "resend" button)
+    # @return [Mail::Message]
+    def shipped_email(options)
+      @order = options.fetch(:order)
+      @carton = options.fetch(:carton)
+      @manifest = @carton.manifest_for_order(@order)
+
+      # Here you can add your custom code to calculate the ETA of the package:
+      @eta = DateTime.now + 1.day
+
+      options = { resend: false }.merge(options)
+      @store = @order.store
+      subject = (options[:resend] ? "[#{t('spree.resend').upcase}] " : '')
+      subject += "#{@store.name} #{t('spree.shipment_mailer.shipped_email.subject')} ##{@order.number}"
+      mail(to: @order.email, from: from_address(@store), subject: subject)
+    end
+  end
+end
+```
+
+Now we need to configure Solidus to use this new mailer instead of the default one:
+
+```ruby title="config/initializers/spree.rb"
+
+Spree.config do |config|
+  config.carton_shipped_email_class = 'AmazingStore::CartonMailer'
+end
+```
+
+This is the list of all the extension points available in Solidus today:
+
+- `order_mailer_class` which defaults to [Spree::OrderMailer][OrderMailer]
+- `carton_shipped_email_class` which defaults to [Spree::CartonMailer][CartonMailer]
+- `reimbursement_mailer_class` which defaults to [Spree::ReimbursementMailer][ReimbursementMailer]
+- `promotion_code_batch_mailer_class` which defaults to [Spree::PromotionCodeBatchMailer][PromotionCodeBatchMailer]
+
+[OrderMailer]: https://github.com/solidusio/solidus/tree/master/core/app/mailers/spree/order_mailer.rb
+[CartonMailer]: https://github.com/solidusio/solidus/tree/master/core/app/mailers/spree/carton_mailer.rb
+[ReimbursementMailer]: https://github.com/solidusio/solidus/tree/master/core/app/mailers/spree/reimbursement_mailer.rb
+[PromotionCodeBatchMailer]: https://github.com/solidusio/solidus/tree/master/core/app/mailers/spre/promotion_code_batch_mailer.rb
+
+### Customize the email template
+
+Next step is adding this new information somewhere in the actual email template, which
+represents what your customers will see in their inbox. The default templates for the
+original carton mailers in Solidus are located at
+[`core/app/views/spree/carton_mailer/`](https://github.com/solidusio/solidus/tree/935c8aeaaa2e427b6ea8f53d220b99aef2d98bf3/core/app/views/spree/carton_mailer).
```

Comment:
```suggestion
[`core/app/views/spree/carton_mailer/`](https://github.com/solidusio/solidus/tree/master/core/app/views/spree/carton_mailer).
```

Reply:
I intentionally used an exact reference here because that folder could be moved in the future. But yes, if that happens, the guide will be out of date so better to notice it as soon as possible.

Reply:
Yeah. I thought that as long as the edge guides track master, we should use links to it.